src/chainparams.cpp

```diff
@@ -249,13 +252,86 @@ class CTestNetParams : public CChainParams {
     }
 };
 
+/**
+ * SigNet
+ */
+class SigNetParams : public CChainParams {
+public:
+    explicit SigNetParams(const ArgsManager& args) {
+        extern CScript g_signet_blockscript;
+
+        std::vector<uint8_t> bin;
+        vSeeds.clear();
+
+        if (!args.IsArgSet("-signet_blockscript")) {
+            throw std::runtime_error(strprintf("%s: -signet_blockscript is mandatory for signet networks", __func__));
+        }
+        if (args.GetArgs("-signet_blockscript").size() != 1) {
+            throw std::runtime_error(strprintf("%s: -signet_blockscript cannot be multiple values.", __func__));
+        }
+        bin = ParseHex(args.GetArgs("-signet_blockscript")[0]);
+        if (args.IsArgSet("-signet_seednode")) {
+            vSeeds = gArgs.GetArgs("-signet_seednode");
```

Comment:
nit: I guess starting a custom signet without a seed node only makes sense if the user is running the seed node themselves. It might be worth a warning or a log message in case people wonder why nothing is happening if they forget this.

Reply:
I think it will be fairly common to start a node and then manually `addnode=` peers to build a network real quick. Having a seed node is only really meant for super permanent signets (like the default one) where you want people to just be able to start and go. I'm not sure how this will all pan out, but that's how I envision it right now.

Reply:
It may still be good to change the output if `-seednode` is passed and echo it rather than only "Using default signet network"

Reply:
nit: `args` not `gArgs`